docs/proposal/20231109-powervs-infra-creation.md

```diff
@@ -0,0 +1,128 @@
+	// Power VS service is a container for all Power VS instances at a specific geographic region.
+	// serviceInstance can be created via IBM Cloud catalog or CLI.
+	// supported serviceInstance identifier in PowerVSResource are Name and ID and that can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// More detail about Power VS service instance.
+	// https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-creating-power-virtual-server
+	// when omitted system will dynamically create the service instance
+	// +optional
+	ServiceInstance IBMPowerVSResourceReference `json:"serviceInstance"`
+
+	// zone is the name of Power VS zone where the cluster will be created
+	Zone string `json:"zone"`
+
+	// resourceGroup name under which the resources will be created.
+	ResourceGroup string `json:"resourceGroup"`
+
+	// vpc contains information about IBM Cloud VPC resources
+	// +optional
+	VPC VPCResource `json:"vpc"`
+
+	// transitGateway contains information about IBM Cloud TransitGateway.
+	// +optional
+	TransitGateway TransitGateway `json:"transitGateway"`
+
+	// controlPlaneLoadBalancer is optional configuration for customizing control plane behavior.
+	// +optional
+	ControlPlaneLoadBalancer *VPCLoadBalancerSpec `json:"controlPlaneLoadBalancer,omitempty"`
+}
+
+
+// IBMPowerVSClusterStatus defines the observed state of IBMPowerVSCluster.
+type IBMPowerVSClusterStatus struct {
+	// ready is true when the provider resource is ready.
+	Ready bool `json:"ready"`
+
+	// serviceInstanceID is the reference to the Power VS service on which the server instance(VM) will be created.
+	ServiceInstanceID *string `json:"serviceInstanceID,omitempty"`
+
+	// networkID is the reference to the Power VS network to use for this cluster.
+	NetworkID *string `json:"networkID,omitempty"`
+
+	// vpcID is reference to IBM Cloud VPC resources.
+	VpcID *string `json:"vpcID,omitempty"`
+
+	// vpcSubnetID is reference to IBM Cloud VPC subnet.
+	VPCSubnetID *string `json:"vpcSubnetID,omitempty"`
+
+	// transitGatewayID is reference to IBM Cloud TransitGateway.
+	TransitGatewayID *string `json:"transitGatewayID,omitempty"`
+
+	// ControlPlaneLoadBalancer reference to IBM Cloud VPC Loadbalancer.
+	ControlPlaneLoadBalancer *VPCLoadBalancerStatus `json:"controlPlaneLoadBalancer,omitempty"`
+
+	// Conditions defines current service state of the IBMPowerVSCluster.
+	Conditions capiv1beta1.Conditions `json:"conditions,omitempty"`
+}
+
+
+// VPCResource holds the VPC information.
+type VPCResource struct {
+	// Name of VPC
+	// +optional
+	Name string `json:"name"`
+
+	// IBM Cloud VPC zone where the cluster's loadbalancers will be created
+	Zone string `json:"vpcZone"`
+
+	// Subnet contains the details about subnet
+	Subnet Subnet `json:"subnet"`
+}
+
+type TransitGateway struct {
+	Name *string `json:"name,omitempty"`
+	ID   *string `json:"id,omitempty"`
+}
+```
+### Following resources will be created
+1. Power VS workspace
+2. Power VS Network [DHCP service]
+3. VPC :
```

Comment:
```suggestion
3. VPC
```
minor nit

+ @Karthik-K-N also please update the proposed API changes doc with final changes as discussed (fields marked optional, omitempty)

Reply:
+ also add `omitempty` marker to all the newly introduced optional fields to maintain consistency (same applies for the other PR)

Reply:
Once the API review PR looks good will copy over the data structure here to be on sych, Does this work?

Reply:
sounds good to me.